### DISCUSSION TITLE
SB campaigns are lost when generating a report via api

### DISCUSSION THREAD

**Discussion by mainefol** [amzn/ads-advanced-tools-docs#246] [Q&A]
Faced a situation where when generating a report via sending an api request, I get a report with not all the campaigns I expect.

While when manually generating reports on the advertising cabinet, all the necessary campaigns are present.

Has anyone encountered something like this? Maybe there are some solutions for this?

**Comment by elizafarley** [comment]
Hi @mainefol, which reporting endpoint are you using? It would be helpful if you could provide a sample request body as well.

**Reply by mainefol** [reply]
Endpoint: https://advertising-api.amazon.com/reporting/reports
Request Body:
ReportRequestBody(
name=SB Campaign Daily report, 
startDate=2024-04-01, 
endDate=2024-04-07, 
configuration=AsyncReportConfiguration(adProduct=SPONSORED_BRANDS, columns=[...columns], groupBy=[campaign], reportTypeId=sbCampaigns, format=GZIP_JSON, timeUnit=DAILY, filters=null)
)

**Reply by elizafarley** [reply]
For version 3 reporting (POST /reporting/reports), some legacy SB campaigns are not supported. Version 3 reporting only returns SB version 4 (multi-ad group supported campaigns). You can check if you have any legacy campaigns by calling [POST /sb/v4/campaigns/list](https://advertising.amazon.com/API/docs/en-us/sponsored-brands/3-0/openapi/prod#tag/Campaigns/operation/ListSponsoredBrandsCampaigns) and seeing if `isMultiAdGroupsEnabled` is set to `false`. Only campaigns with `isMultiAdGroupsEnabled` is set to `true` are returned in version 3 reporting. 

If you need data for legacy SB campaigns, you will need to use [version 2 reporting](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v2/report-types).
